### PR TITLE
[docs] update to expose libraries + landing page

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

This reorganizes the tune docs to give important algorithms their own page (we should add more as new algorithms are added). This should hopefully help with SEO -- currently if you search for "HyperBand" the tune docs don't even show up in the first couple pages. However if you search for "A3C" or "Rllib" those terms already rank highly.

lbs

### CODE REVIEW

**Comment by richardliaw** [comment]
Somehow it might be nice to indicate that HyperBand isn't a library, as with PBT. I do believe these things are worth highlighting and am not sure about the right formatting.

@robertnishihara @pcmoritz any thoughts?

**Comment by ericl** [comment]
Yeah I don't know if readthedocs has some syntax for a nested page.

**Comment by richardliaw** [comment]
screenshot 2018-03-06 17 48 13

**Comment by robertnishihara** [comment]
seems good to me

**Review by richardliaw — APPROVED** [review]
@ericl, should do a read over my changes before merge

File: doc/source/hyperband.rst [hunk 1]
```diff
@@ -50,9 +50,40 @@ Note that the HyperBand scheduler requires your trainable to support checkpointi
...
+.. image:: images/hyperband_allocation.png
+
+
+3. There are also implementation specific details like how trials are placed into brackets which are not covered in the paper. Our implementation places trials within brackets according to smaller bracket first - meaning that with low number of trials, there will be less early stopping.
+
+
+Asynchronous HyperBand
+----------------------
+
+Ray Tune also implements an `asynchronous version of HyperBand <https://openreview.net/forum?id=S1Y7OOlRZ>`__, providing better parallelism and avoids straggler issues during eliminations. An example of this can be found in `async_hyperband_example.py <https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/async_hyperband_example.py>`__. **We recommend using this over the vanilla HyperBand scheduler.**
```
Comment (ericl) [inline doc/source/hyperband.rst]: Should we recommend this as the default one?

File: doc/source/hyperband.rst [hunk 2]
```diff
@@ -50,9 +50,40 @@ Note that the HyperBand scheduler requires your trainable to support checkpointi
...
+
+1. Both `s_max` (representing the ``number of brackets - 1``) and `eta`, representing the downsampling rate, are fixed.  In many practical settings, you may have `R >= 200`. For simplicity, assume `eta = 3`. Between `R = 200` and `R = 1000`, you end up with a huge range of the number trials needed to fill up all brackets.
+
+.. image:: images/hyperband_bracket.png
+
+Now, assume `R = 300`. If you want to find a better `eta`, you realize also that the configurations generated aren't very intuitive:
+
+.. image:: images/hyperband_eta.png
+
+We figured that it would be a lot easier for majority of people to use if we fixed the parameters to be the same as the example given in the paper and only exposed `max_t`.
```
Comment (ericl) [inline doc/source/hyperband.rst]: This phrasing sounds more like a blog post, reword for documentation?

**Review by ericl — COMMENTED** [review]
Looks good, couple comments on phrasing

**Comment by richardliaw** [comment]
addressed comments and tweaked a little more; I'll merge for now after tests pass(as there are no substantial changes) but if there are issues we can just patch it.

**Pull request merged**